File: Java/stararrangements.java

// https://open.kattis.com/problems/stararrangements
import java.util.Scanner;

public class stararrangements {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double stars = sc.nextDouble();
        System.out.println((int) stars + ":");
        for (int x = 2; x < Math.round(stars / 2) + 1; x++) {
            int y = x - 1;
            if (stars % (x + y) == x || stars % (x + y) == 0)
                System.out.println(x + "," + y);
            if (stars % x == 0)
                System.out.println(x + "," + x);
        }
        sc.close();
    }
}

File: C#/stararrangements.cs
using System;

public class stararrangements {
    public static void Main (string[] args) {
        int stars = int.Parse (Console.ReadLine ());
        Console.WriteLine (stars + ":");
        for (int x = 2; x <= (stars / 2) + 1; x++) {
            if (stars % (x + (x - 1)) == x || stars % (x + (x - 1)) == 0)
                Console.WriteLine (x + "," + (x - 1));
            if (stars % x == 0)
                Console.WriteLine (x + "," + x);
        }
    }
}